lake/branch.go

```diff
@@ -174,10 +183,49 @@ func (b *Branch) DeleteWhere(ctx context.Context, c runtime.Compiler, program as
 			obj := o
 			patch.AddDataObject(&obj)
 		}
+		if message == "" {
+			var deletedObjs []data.Object
```

Comment:
Nit: No point in copying these.
```suggestion
			var deletedObjs []*data.Object
```